src/test/java/com/artipie/asto/StorageExtension.java

```diff
@@ -78,21 +108,33 @@ public boolean supportsTestTemplate(final ExtensionContext context) {
         final ExtensionContext context) {
         final Collection<Storage> storages;
         try {
-            storages = Arrays.asList(
-                new InMemoryStorage(),
-                this.s3Storage(),
-                new SubStorage(new Key.From("mem-prefix"), new InMemoryStorage()),
-                new SubStorage(
-                    new Key.From("file-prefix"),
-                    new FileStorage(Files.createTempDirectory("pref-sub"))
-                ),
-                new SubStorage(Key.ROOT, new InMemoryStorage()),
-                new SubStorage(Key.ROOT, new FileStorage(Files.createTempDirectory("sub"))),
-                new FileStorage(Files.createTempDirectory("junit")),
-                new VertxFileStorage(Files.createTempDirectory("vtxjunit"), this.vertx),
-                new BenchmarkStorage(new InMemoryStorage())
+            storages = new LinkedList<>(
+                Arrays.asList(
+                    new InMemoryStorage(),
+                    this.s3Storage(),
+                    new SubStorage(new Key.From("mem-prefix"), new InMemoryStorage()),
+                    new SubStorage(
+                        new Key.From("file-prefix"),
+                        new FileStorage(Files.createTempDirectory("pref-sub"))
+                    ),
+                    new SubStorage(Key.ROOT, new InMemoryStorage()),
+                    new SubStorage(Key.ROOT, new FileStorage(Files.createTempDirectory("sub"))),
+                    new FileStorage(Files.createTempDirectory("junit")),
+                    new VertxFileStorage(Files.createTempDirectory("vtxjunit"), this.vertx),
+                    new BenchmarkStorage(new InMemoryStorage())
+                )
             );
-        } catch (final IOException ex) {
+            if (!SystemUtils.IS_OS_WINDOWS) {
+                storages.add(
```

Comment:
I think we need to check if etcd storage works fine with `SubStorage` (like other storahes are checked): `new SubStorage(someKey, etcdStorage)`

Reply:
I see